fhir-server/src/main/java/au/csiro/pathling/QueryExecutor.java

```diff
@@ -70,8 +71,10 @@ protected QueryExecutor(@Nonnull final Configuration configuration,
   @Nonnull
   protected ParserContext buildParserContext(@Nonnull final ResourceType resourceType) {
     final ResourcePath inputContext = buildInputContext(resourceType);
-    return new ParserContext(inputContext, Optional.empty(), fhirContext, sparkSession,
-        resourceReader, terminologyClient, terminologyClientFactory);
+    check(inputContext.getIdColumn().isPresent());
```

Comment:
This seem's to be outside of control of the user/caller and a violation most likely indicates a bug in the core rather than an illegal argument, so perhaps could be better expressed with `assert`?

Reply:
I actually have a `checkPresent` method which I should be using here, so I'll use that. It uses `Optional.orElseThrow()`.

We were previously using assertions quite liberally, but I wanted to move away from that. The reason for that was that we don't always have control over the execution environment that Pathling is deployed within, and assertions are not always enabled within the JVM.

Reply:
What I have done is change the type of exception thrown by `Preconditions.check` to be an `AssertionError`, to better fit as a replacement for `assert`.

I've also added this new method that throws an `IllegalArgumentException`, and also accepts an error message:

```java
  /**
   * Ensures the truth of an expression, throwing an {@link IllegalArgumentException} with the
   * supplied message if it does not evaluate as true.
   *
   * @param expression The expression that should be true
   * @param errorMessage The message to use if an error is thrown
   */
  public static void checkArgument(final boolean expression, @Nonnull final String errorMessage) {
    if (!expression) {
      throw new IllegalArgumentException(errorMessage);
    }
  }
```

Reply:
See b90fdb5850f3476a99eddff7e81979134cac03e5.